modules/client/src/internal/client/methods.ts

```diff
@@ -1075,4 +1084,69 @@ export class ClientMethods extends ClientCore implements IClientMethods {
     }
     return version;
   }
+
+  public async isDaoUpdateProposalValid(
+    proposalId: string,
+  ): Promise<DaoUpdateProposalValidity> {
+    if (!isProposalId(proposalId)) {
+      throw new InvalidProposalIdError();
+    }
+    // search for the proposal
+    const name = "IProposal";
+    const query = QueryIProposal;
+    type T = { iproposal: SubgraphIProposal };
+    const { iproposal } = await this.graphql.request<T>({
+      query,
+      params: { id: proposalId.toLowerCase() },
+      name,
+    });
+    if (!iproposal) {
+      throw new ProposalNotFoundError();
+    }
+    const res: DaoUpdateProposalValidity = {
+      version: false,
+      implementation: false,
+      initData: false,
+      actions: false,
+    };
+    // check if the proposal has only one action for upgrading the dao
+    if (iproposal.actions.length === 1) {
+      const action = iproposal.actions[0];
+      try {
+        const upgradeToAdCallParams = decodeUpgradeToAndCallAction(
+          hexToBytes(action.data),
+        );
+        const decodedInitializeFromParams = decodeInitializeFromAction(
+          upgradeToAdCallParams.data,
+        );
+        // action is decoded son the actions are valid
+        res.actions = true
+        const currentDaoVersion = await this.getProtocolVersion(
+          iproposal.dao.id,
+        );
+        // check if the version of the specified dao matches the version in the encoded data
+        if (
+          JSON.stringify(currentDaoVersion) ===
+            JSON.stringify(decodedInitializeFromParams.previousVersion)
+        ) {
+          res.version = true;
+        }
+        // check if the implementation matches the latest daoFactoryAddress
+        if (
+          this.web3.getAddress("daoFactoryAddress") ===
+            upgradeToAdCallParams.implementationAddress
+        ) {
+          res.implementation = true;
+        }
```

Comment:
Don't u have to the contract of `daoFactory` then get the implementation (`daoBase`) then compare it to `upgradeToAdCallParams.implementationAddress`?

Reply:
Maybe I didnt fully understood the reuirement then, why does de daoBase os the original implmentation has to be equal to the new implementation?

Reply:
The DAOFactory is the contract that creates new DAOs and it holds a public constant called `daoBase` which it uses to create the new DAO. Thus an upgrade should point to the `daoBase` and not the DAOFactory

Reply:
if u look at the daoFactory's code u see it deploy a proxy and set daoBase as implementation
so we needed to compare update's new implem. to the one on the current daoFactory?